tsl/src/continuous_aggs/create.c

```diff
@@ -1173,39 +1183,148 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 				 detail->len > 0 ? errdetail("%s", detail->data) : 0));
 	}
 
-	/* finalized cagg doesn't have those restrictions anymore */
+	/* Finalized cagg doesn't have those restrictions anymore */
 	if (!finalized)
 	{
 		/* validate aggregates allowed */
 		cagg_agg_validate((Node *) query->targetList, NULL);
 		cagg_agg_validate((Node *) query->havingQual, NULL);
 	}
-
+	/* Check if there are only two tables in the from list */
 	fromList = query->jointree->fromlist;
-	if (list_length(fromList) != 1 || !IsA(linitial(fromList), RangeTblRef))
+	if (list_length(fromList) > 2)
```

Comment:
```suggestion
	#define CAGG_MAX_JOIN_RELATIONS -- put this at the beginning of the file together with other defines
	if (list_length(fromList) > CAGG_MAX_JOIN_RELATIONS)
```
We're using `2` in several places so it will make the code more readable!